src/compartment.c

```diff
@@ -183,20 +188,22 @@ comp_from_elf(char* filename, char** entry_points)
 
             size_t syms_count = comp_symtb_hdr.sh_size / sizeof(Elf64_Sym);
             Elf64_Sym curr_sym;
-            char** syms_to_find = entry_points;
-            size_t syms_to_find_count = sizeof(syms_to_find) / sizeof(syms_to_find[0]);
-            char** syms_found = malloc(syms_to_find_count);
+            size_t syms_to_find_count = entry_point_count;
+            const char** syms_found = malloc(syms_to_find_count);
             size_t syms_found_count = 0;
             for (size_t j = 0; j < syms_count; ++j)
             {
                 curr_sym = comp_symtb[j];
                 for (size_t k = 0; k < syms_to_find_count; ++k)
                 {
-                    if (!strcmp(syms_to_find[k], &comp_strtb[curr_sym.st_name])) // TODO entry point name
+                    if (!strcmp(entry_points[k].name, &comp_strtb[curr_sym.st_name])) // TODO entry point name
                     {
                         struct entry_point* new_entry_point = malloc(sizeof(struct entry_point));
-                        new_entry_point->fn_name = malloc(strlen(syms_to_find[k]) + 1);
-                        strcpy(new_entry_point->fn_name, syms_to_find[k]);
+                        new_entry_point->fn_name = entry_points[k].name;
+                        new_entry_point->arg_count = entry_points[k].arg_count;
+                        new_entry_point->arg_types = entry_points[k].args_type;
+                        /*new_entry_point->fn_name = malloc(strlen(syms_to_find[k]) + 1);*/
```

Comment:
Commented code?

Reply:
Removed in https://github.com/capablevms/CHERI-ELF-comp/pull/7/commits/d5e6106c5f6a661414c66aacc2a361cfc93b0f4b.